packages/plugin-git/sources/gitUtils.ts

```diff
@@ -170,12 +170,17 @@ export function normalizeLocator(locator: Locator) {
 }
 
 export async function lsRemote(repo: string, configuration: Configuration) {
-  if (!configuration.get(`enableNetwork`))
+  const normalizedRepoUrl = normalizeRepoUrl(repo, {git: true});
+
+  if (
+    !configuration.get(`enableNetwork`) ||
+    httpUtils.getNetworkSettings(normalizedRepoUrl, {configuration})?.config.get(`enableNetwork`) === false
```

Comment:
Feels like `httpUtils.getNetworkSettings` should also do the `configuration.get('enableNetwork')` check.